tests/sentry/search/events/test_filter.py

```diff
@@ -902,11 +902,20 @@ def test_wildcard_array_field(self):
         assert _filter.filter_keys == {}
 
     def test_wildcard_array_field_with_backslash(self):
-        _filter = get_filter(r"stack.filename:\k*")
-        assert _filter.conditions == [
-            ["stack.filename", "LIKE", "\\\\k%"],
+        test_cases = [
+            (r"stack.filename:\k*", ["stack.filename", "LIKE", "\\\\k%"]),  # prefixed by \k
+            (r"stack.filename:\\*", ["stack.filename", "LIKE", "\\\\\\\\%"]),  # prefixed by \\
+            (r"stack.filename:\**", ["stack.filename", "LIKE", "\\\\%%"]),  # prefixed by \%
```

Comment:
This test case is the current behaviour but is actually wrong right? Since the first asterisks should be escaped by the backslash meaning the query should be `"*%"` where the first asterisk is a literal and the second one is a wildcard. Can we leave a comment noting that so the next person isn't confused by why this is the expected behaviour?